aioxmpp/blocking/service.py

```diff
@@ -0,0 +1,225 @@
+
+    @asyncio.coroutine
+    def _get_initial_blocklist(self):
+        yield from self._check_for_blocking()
+
+        if self._blocklist is None:
+            with (yield from self._lock):
+                iq = aioxmpp.IQ(
+                    type_=aioxmpp.IQType.GET,
+                    payload=blocking_xso.BlockList(),
+                )
+                result = yield from self.client.stream.send(iq)
+                self._blocklist = frozenset(result.items)
+            self.on_initial_blocklist_received(self._blocklist)
+
+    @property
+    def blocklist(self):
+        """
+        :class:`~collections.abc.Set` of JIDs blocked by the account.
+        """
+        return self._blocklist
+
+    @asyncio.coroutine
+    def block_jids(self, jids_to_block):
+        """
+        Add the JIDs in the sequence `jids_to_block` to the client's
+        blocklist.
+        """
+        yield from self._check_for_blocking()
+
+        if not jids_to_block:
+            return
+
+        cmd = blocking_xso.BlockCommand()
+        cmd.items[:] = jids_to_block
+        iq = aioxmpp.IQ(
+            type_=aioxmpp.IQType.SET,
+            payload=cmd,
+        )
+        yield from self.client.stream.send(iq)
+
+    @asyncio.coroutine
+    def unblock_jids(self, jids_to_unblock):
+        """
+        Remove the JIDs in the sequence `jids_to_block` from the
+        client's blocklist.
+        """
+        yield from self._check_for_blocking()
+
+        if not jids_to_unblock:
+            return
+
+        cmd = blocking_xso.UnblockCommand()
+        cmd.items[:] = jids_to_unblock
+        iq = aioxmpp.IQ(
+            type_=aioxmpp.IQType.SET,
+            payload=cmd,
+        )
+        yield from self.client.stream.send(iq)
+
+    @asyncio.coroutine
+    def unblock_all(self):
+        """
+        Unblock all JIDs currently blocked.
+        """
+        yield from self._check_for_blocking()
+
+        cmd = blocking_xso.UnblockCommand()
+        iq = aioxmpp.IQ(
+            type_=aioxmpp.IQType.SET,
+            payload=cmd,
+        )
+        yield from self.client.stream.send(iq)
+
+    @service.iq_handler(aioxmpp.IQType.SET, blocking_xso.BlockCommand)
+    @asyncio.coroutine
+    def handle_block_push(self, block_command):
+        must_signal = False
+        with (yield from self._lock):
+            if self._blocklist is None:
+                # this means the stream was destroyed while we were waiting for
+                # the lock/while the handler was enqueued for scheduling, or
+                # the server is buggy and sends pushes before we fetched the
+                # blocklist
+                return
+
+            if (block_command.from_ is None or
+                    block_command.from_ == self.client.local_jid.bare()):
+                diff = frozenset(block_command.payload.items)
+                self._blocklist |= diff
+                must_signal = True
+
+        if must_signal:
```

Comment:
Instead of having this boolean, why not set diff to ``()`` at the top and check for ``if diff:``?